flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStStateExecutor.java

```diff
@@ -0,0 +1,129 @@
+     * The number of coordinator threads is 1 to ensure that multiple batches of stateRequests can
+     * be executed sequentially.
+     */
+    private final ExecutorService coordinatorThread;
+
+    /** The worker thread that actually executes the {@link StateRequest}s. */
+    private final ExecutorService workerThreads;
+
+    private final RocksDB db;
+
+    private final WriteOptions writeOptions;
+
+    public ForStStateExecutor(int ioParallelism, RocksDB db, WriteOptions writeOptions) {
+        this.coordinatorThread =
+                Executors.newSingleThreadScheduledExecutor(
+                        new ExecutorThreadFactory("ForSt-StateExecutor-Coordinator"));
+        this.workerThreads =
+                Executors.newFixedThreadPool(
+                        ioParallelism, new ExecutorThreadFactory("ForSt-StateExecutor-IO"));
+        this.db = db;
+        this.writeOptions = writeOptions;
+    }
+
+    @Override
+    public CompletableFuture<Void> executeBatchRequests(
+            StateRequestContainer stateRequestContainer) {
+        Preconditions.checkArgument(stateRequestContainer instanceof ForStStateRequestClassifier);
+        ForStStateRequestClassifier stateRequestClassifier =
+                (ForStStateRequestClassifier) stateRequestContainer;
+        return CompletableFuture.supplyAsync(
+                () -> {
+                    long startTime = System.currentTimeMillis();
+                    List<CompletableFuture<Void>> futures = new ArrayList<>(2);
+                    List<ForStDBPutRequest<?, ?>> putRequests =
+                            stateRequestClassifier.pollDbPutRequests();
+                    if (!putRequests.isEmpty()) {
+                        ForStWriteBatchOperation writeOperations =
+                                new ForStWriteBatchOperation(
+                                        db, putRequests, writeOptions, workerThreads);
+                        futures.add(writeOperations.process());
+                    }
+
+                    List<ForStDBGetRequest<?, ?>> getRequests =
+                            stateRequestClassifier.pollDbGetRequests();
+                    if (!getRequests.isEmpty()) {
+                        ForStGeneralMultiGetOperation getOperations =
+                                new ForStGeneralMultiGetOperation(db, getRequests, workerThreads);
+                        futures.add(getOperations.process());
+                    }
+
+                    try {
+                        FutureUtils.waitForAll(futures).get();
+                    } catch (InterruptedException | ExecutionException e) {
+                        throw new FlinkRuntimeException(
+                                "Exception when executing ForSt writeBatch or multiGet operation",
+                                e);
+                    }
+                    long duration = System.currentTimeMillis() - startTime;
+                    LOG.info(
```

Comment:
should be `debug` ? it may introduce many unnecessary logs especially for small buffer size.

Reply:
resolved.